ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -1514,6 +1523,18 @@ public static void mvFileToFinalPath(Path specPath, Configuration hconf,
     fs.delete(taskTmpPath, true);
   }
 
+  private static void createFileList(Set<FileStatus> filesKept, Path srcPath, Path targetPath, FileSystem fs)
+      throws IOException {
+    String files = srcPath.toString() + System.lineSeparator();
+    for (FileStatus file : filesKept) {
+      files += file.getPath().toString() + System.lineSeparator();
+    }
+    try (FSDataOutputStream outStream = fs.create(new Path(targetPath, BLOB_FILES_KEPT))) {
+      outStream.writeBytes(files);
```

Comment:
Why not loop through filesKept here and directly write them to outStream? That will mem pressure in constructing "files"